microservices/services/query-metric/service/src/main/java/datawave/microservice/querymetric/QueryMetricOperations.java

```diff
@@ -541,6 +541,31 @@ public QueryGeometryResponse map(@AuthenticationPrincipal DatawaveUserDetails cu
         }
     }
     
+    /**
+     * Returns subplans for the current user's query that is identified by the id
+     *
+     * @param currentUser
+     *            the current user
+     * @param queryId
+     *            the query id
+     * @return the ModelAndView for the webpage
+     * @HTTP 200 success
+     * @HTTP 500 internal server error
+     */
+    @Operation(summary = "Get the subplans for a given query ID.")
+    @PermitAll
+    @RequestMapping(path = "/id/{queryId}/subplans", method = {RequestMethod.GET}, produces = {MediaType.TEXT_HTML_VALUE})
+    public ModelAndView querySubplanWebpage(@AuthenticationPrincipal DatawaveUserDetails currentUser,
+                    @Parameter(description = "queryId to return") @PathVariable("queryId") String queryId,
+                    @Parameter(description = "queryId to return") @RequestParam(name = "display", required = false) String display) {
+        
+        BaseQueryMetricListResponse response = query(currentUser, queryId);
+        if (StringUtils.isNotBlank(display) && display.equalsIgnoreCase("horizontal")) {
+            response.setViewName("querysubplans");
+        }
+        return response.createSubplanModelAndView();
+    }
+    
```

Comment:
It is not necessary to have the display parameter here or the conditional logic if you are not supporting multiple modelviews in this operation (like queryWebpage which supports the legacy horizontal html view).

Your viewname is already set in createSubplanModelAndView